src/App.tsx

```diff
@@ -9,39 +9,20 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppSelector } from './app/hooks';
+import { LoadingStatuses } from './enums/LoadingStatuses';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
-
-  useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+  const { idle, loading, error } = LoadingStatuses;
+  const { posts, postsLoadStatus, activePost } = useAppSelector(
+    state => state.posts,
+  );
+  const { author } = useAppSelector(state => state.users);
```

Comment:
It's better to create selectors https://redux.js.org/usage/deriving-data-selectors#basic-selector-concepts instead of passing anonim function every time when you need something from the state